x-pack/plugins/observability/public/components/alerts_table/register_alerts_table_configuration.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AlertTableConfigRegistry } from '@kbn/triggers-actions-ui-plugin/public/application/alert_table_config_registry';
+import { ObservabilityRuleTypeRegistry } from '../..';
+import { getAlertsPageTableConfiguration } from './alerts/get_alerts_page_table_configuration';
+import { getRuleDetailsTableConfiguration } from './rule_details/get_rule_details_table_configuration';
+import { getSloAlertsTableConfiguration } from './slo/get_slo_alerts_table_configuration';
+
+export const registerAlertsTableConfiguration = (
+  alertTableConfigRegistry: AlertTableConfigRegistry,
+  observabilityRuleTypeRegistry: ObservabilityRuleTypeRegistry,
+  config: any
+) => {
+  // Alert page
+  const alertsPageAlertsTableConfig = getAlertsPageTableConfiguration(
+    observabilityRuleTypeRegistry,
+    config
+  );
+  alertTableConfigRegistry.register(alertsPageAlertsTableConfig);
+
+  // Rule details page
+  const ruleDetailsAlertsTableConfig = getRuleDetailsTableConfiguration(
```

Comment:
Nice refactoring doing all alerts table configuration registration in one place. I am just wondering what difference alerts table has in rule details page vs alerts page? Is it the extra `Rule name` column that we don't want it to be visible in the Rule details page, cause it is redundant there? Any other difference I didn't notice?

Reply:
Also this change will affect any other o11y solutions that use the alerts table configuration (apm, could it be cases as well?). Is it fine that the other solutions will get the new default columns?

Reply:
> Is it the extra Rule name column that we don't want it to be visible in the Rule details page, cause it is redundant there? 

Yes, exactly, that's the only difference at the moment.

>Also this change will affect any other o11y solutions that use the alerts table configuration (apm, could it be cases as well?). Is it fine that the other solutions will get the new default columns?

Since I changed the tests for the infra team, they are also notified, I also see @smith added `apm:review`, so I'll wait for their approval as well.